taxi/views.py

```diff
@@ -61,17 +65,55 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def get_queryset(self):
+        return (
+            super().get_queryset()
+            .select_related("manufacturer")
+            .prefetch_related(
+                Prefetch(
+                    "drivers",
+                    queryset=get_user_model().objects.only(
+                        "username",
+                        "first_name",
+                        "last_name"
+                    ),
+                )
+            )
+        )
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["drivers"] = self.object.drivers.all()
+
+        return context
+
+
+@login_required
```

Comment:
Include this logic into post method of CarDetailView. This way you don't need to create extra view and url endpoint to handle adding/deleting driver.
So the implementation will look something like this:

1. Update form on car detail template that will send value 'Add' or 'Delete' to post method to the cardetailview.
2. You override post method in this view and catch those values that you received in if condition and add/remove driver to car through related fields.
3. You will need to figure out how to get car and user in the view's post method to set related fields.